aries_cloudagent/resolver/routes.py

```diff
@@ -0,0 +1,119 @@
+"""resolve did document admin routes."""
+
+
+import re
+from aiohttp import web
+from aiohttp_apispec import (
+    docs,
+    match_info_schema,
+    response_schema,
+)
+from marshmallow import fields, validate
+
+from ..admin.request_context import AdminRequestContext
+from ..messaging.models.base import BaseModelError
+from ..messaging.models.openapi import OpenAPISchema
+from ..storage.error import StorageError, StorageNotFoundError
+from .did_resolver import DIDResolver
+
+
+class W3cDIDDoc(validate.Regexp):
+    """Validate value against w3c DID document."""
+
+    EXAMPLE = "*"
+    PATTERN = re.compile(r"[a-zA-Z0-9._-]*")
+
+    def __init__(self):
+        """Initializer."""
+
+        super().__init__(
+            W3cDIDDoc.PATTERN,
+            error="Value {input} is not a w3c decentralized identifier (DID) doc.",
+        )
+
+
+_W3cDIDDoc = {"validate": W3cDIDDoc(), "example": W3cDIDDoc.EXAMPLE}
+
+
+class DIDDocSchema(OpenAPISchema):
+    """Result schema for did document query."""
+
+    did_doc = fields.Str(
+        description="decentralize identifier(DID) document", required=True, **_W3cDIDDoc
+    )
+
+
+class W3cDID(validate.Regexp):
+    """Validate value against w3c DID."""
+
+    EXAMPLE = "did:ted:WgWxqztrNooG92RXvxSTWv"
+    PATTERN = re.compile(r"did:([a-z]+):((?:[a-zA-Z0-9._-]*:)*[a-zA-Z0-9._-]+)")
```

Comment:
We should reuse the regex in the DID file. Should be accessible with `from .did import DID_PATTERN`